src/MOIWrapper.jl

```diff
@@ -0,0 +1,373 @@
+export SCSOptimizer
+
+using MathOptInterface
+const MOI = MathOptInterface
+const CI = MOI.ConstraintIndex
+const VI = MOI.VariableIndex
+
+const MOIU = MOI.Utilities
+
+const SF = Union{MOI.SingleVariable, MOI.ScalarAffineFunction{Float64}, MOI.VectorOfVariables, MOI.VectorAffineFunction{Float64}}
+const SS = Union{MOI.EqualTo{Float64}, MOI.GreaterThan{Float64}, MOI.LessThan{Float64}, MOI.Zeros, MOI.Nonnegatives, MOI.Nonpositives, MOI.SecondOrderCone, MOI.ExponentialCone, MOI.PositiveSemidefiniteConeTriangle}
+
+struct MOISolution
+    ret_val::Int
+    primal::Vector{Float64}
+    dual::Vector{Float64}
+    slack::Vector{Float64}
+    objval::Float64
+end
+MOISolution() = MOISolution(0, # SCS_UNFINISHED
+                      Float64[], Float64[], Float64[], NaN)
+
+mutable struct Data
+    m::Int
+    n::Int
+    I::Vector{Int}
+    J::Vector{Int}
+    V::Vector{Float64}
+    b::Vector{Float64}
+    objconstant::Float64
+    c::Vector{Float64}
+end
+
+mutable struct Cone
+    f::Int # number of linear equality constraints
+    l::Int # length of LP cone
+    q::Int # length of SOC cone
+    qa::Vector{Int} # array of second-order cone constraints
+    s::Int # length of SD cone
+    sa::Vector{Int} # array of semi-definite constraints
+    ep::Int # number of primal exponential cone triples
+    ed::Int # number of dual exponential cone triples
+    p::Vector{Float64} # array of power cone params
+    setconstant::Dict{Int, Float64} # For the constant of EqualTo, LessThan and GreaterThan
```

Comment:
I don't understand what this does. The rest of the struct is simply SCS's cone format, so `setconstant` and `nrows` need more description.